src/test/java/com/puppycrawl/tools/checkstyle/api/XpathMapperTest.java

```diff
@@ -0,0 +1,151 @@
+package com.puppycrawl.tools.checkstyle.api;
+
+import antlr.RecognitionException;
+import antlr.TokenStreamException;
+import com.puppycrawl.tools.checkstyle.BaseCheckTestSupport;
+import com.puppycrawl.tools.checkstyle.TreeWalker;
+import com.puppycrawl.tools.checkstyle.xpath.ElementNode;
+import com.puppycrawl.tools.checkstyle.xpath.RootNode;
+import net.sf.saxon.om.GenericTreeInfo;
+import net.sf.saxon.om.Item;
+import net.sf.saxon.sxpath.XPathDynamicContext;
+import net.sf.saxon.sxpath.XPathEvaluator;
+import net.sf.saxon.sxpath.XPathExpression;
+import net.sf.saxon.trans.XPathException;
+import org.junit.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.List;
+
+import static org.junit.Assert.assertArrayEquals;
+
+/**
+ * Created by timur on 6/13/17.
+ */
+public class XpathMapperTest extends BaseCheckTestSupport {
+
+    @Test
+    public void testOne() throws XPathException, IOException, TokenStreamException, RecognitionException {
+        final FileText text = new FileText(new File("src/test/resources/com/puppycrawl/tools/checkstyle/api/TestAst.java"),
+                System.getProperty("file.encoding", "UTF-8"));
+        final FileContents contents = new FileContents(text);
+
+        DetailAST rootAST = TreeWalker.parse(contents);
+        DetailNode detailNode = null;
+
+        RootNode rootNode = new RootNode();
+        ElementNode classDefinition = new ElementNode(TokenTypes.CLASS_DEF, rootNode, rootNode, detailNode, rootAST.getNextSibling());
```

Comment:
I don't really understand how DetailNode relate to DetalAST, is it one of the possible children?